basics/task1-6/print_data.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+require 'email_validation.php';
+
+/**
+ * Prints the user data on the webpage
+ *
+ * @param string $full_name
+ *   User's full name.
+ * @param string $email_address
+ *   User's email address.
+ * @param string $number
+ *   User's contact number.
+ * @param mixed $image
+ *   User's input image.
+ * @param string $marks
+ *   User's subject marks.
+ */
+function show_data(string $full_name, string $email_address, string $number, mixed $image, string $marks): void {
+  ?>
+  <h1>Hello
+    <?= $full_name ?>
+  </h1>
+  <img src='<?= $image ?>' height='200px'>
+  <h2>Subject Marks</h2>
+  <table border='1px'>
+    <tr>
+      <th>Subject</th>
+      <th>Mark</th>
+    </tr>
+    <?php
+    $marks = explode("\n", $marks);
+    foreach ($marks as $mark) {
+      $subject_mark = explode("|", $mark);
+      $subject = isset($subject_mark[0]) ? trim($subject_mark[0]) : '';
+      $mark = isset($subject_mark[1]) ? trim($subject_mark[1]) : '';
+      ?>
+      <tr>
+        <td>
+          <?= $subject ?>
+        </td>
+        <td>
+          <?= $mark ?>
+        </td>
+      </tr>
+      <?php
+    }
+    ?>
+  </table>
+  <p>Phone Number:
+    <?= $number ?>
+  </p>
+  <?php
+
+  //Checks if the input email address is valid or not.
+  $valid = isEmailValid($email_address);
```

Comment:
Why are we displaying whether an email is valid or invalid on the output page? Ideally, we should validate all fields, such as name, phone number, email, subject, marks, etc., first. If all validations pass, then we proceed to display the output. Otherwise, we show errors alongside their respective fields in the form itself.